src/reporter.ts

```diff
@@ -282,6 +282,15 @@ export class RPReporter implements Reporter {
       mode: mode || LAUNCH_MODES.DEFAULT,
       id: launchId,
     };
+
+    // Extract grep and grepInvert from config and add as launch attributes
+    if (this.config.grep) {
+      startLaunchObj.attributes.push({ key: 'grep', value: this.config.grep });
+    }
+    if (this.config.grepInvert) {
+      startLaunchObj.attributes.push({ key: 'grepInvert', value: this.config.grepInvert });
+    }
```

Comment:
Use [config: FullConfig argument of onBegin method](https://playwright.dev/docs/test-reporters#custom-reporters:~:text=implements%20Reporter%20%7B-,onBegin) to check if the `grep` or `grepInvert` options provided.